### ISSUE TITLE
Troughs and buckets are unweildy, but kind of fun also, precision mode pour?

### ISSUE THREAD

**Issue by GreenboxGames** [naelstrof/KoboldKare#158]
I can't aim them as it seems like you yeet at a random angle from your crosshair (er, dot). Sometimes it goes pretty much forward but other times it goes almost straight up, or at least like a 45 angle to the crosshair. I noticed that when I shift (hand grab mode) the hand can appear on the container I want to empty into. Maybe have it focus the splash, or even just auto fill the container the hand is on when pressing the grab mode key? I still think splashing wildly can be fun, especially in multiplayer, but some precision would be nice as I spill a lot when trying to transfer to the watering cans.

**Comment by naelstrof** [comment]
Yeah this is a hard thing to tackle, it going wildly upwards is due to it inheriting the velocity of the player. The player's vertical velocity has really really strong forces on it to make it walk smootly on the ground. The farm plots cause the PID to push you up a bit and wobble-- which is usually the cause of the bucket contents flying up.

I could make it ignore the y velocity but then it gets weird because walking up and down hills cause it to act inappropriately. I didn't really figure out a good solution, I made it just inherit half of the velocity and it helped though. I'm open to solutions.

A precision mode seems difficult to implement, and was intended to be done with the watering can. Is there a reason why the watering can doesn't serve purpose that you describe?

**Comment by GreenboxGames** [comment]
I did a crap job of explaining what I was actually doing. Mostly just using the multiplayer thing to justify cum bucket battles and forgot the point I was making, lol. I'm not watering Kobolds but doing recipes trying to fill the watering cans. I'm having trouble with this sitting still, on a flat surface, while trying to fill the watering cans and do precision alchemy, no velocity should be present, or is purposely applied on my Kobold's transform. I would assume that you would want it to take into account the actual container's velocity as containers would, I assume, be naturally inheriting the velocity of the carrier (physics body component) via physics engine and object hierarchy, but you could be doing something different from how I'm thinking in my head. I'll have to actually look at your implementation today to say for sure. Yesterday I did just learn to fill fruit and make recipes in the watering cans almost exclusively to distribute things better so if there are better-intended methods to what I will describe let me know. My intuition could be wonky.

The watering cans are great for precision work for sure, I'm wishing that everything kinda worked that way. What I'm doing is things like filling the 60-count troughs with milk after pie, which works better for me than the milking machine since you can fill two at once and move yourself to two others rather quickly or have a chance to move them with another Kobold while one is filling... or maybe I just like it better? It seems sexier than having to terminate the intimacy of milking and then run around trying to swap troughs in front of the milk machine, which takes some getting used to. Thank goodness containers don't spill, but would make for an interesting hardcore mode rule.

In my experience the troughs splash out at 15 units a time, I think 3 chunks of 5 units. I think the buckets are 2 chunks 5 units? It's hard to get that into a watering can (like I say the angles seem random, sitting still on a flat surface for me) but is perfectly half and a third watering can capacity, but would be nice if I could fill in 5 unit bits, dipping the watering can in the trough for 5 units at a time maybe?

Now if I want to fill with cum I have the injector that works well by locking the watering cans (or anything) above the dongnozzle and pumping it. That works well. (Come to think of it, I haven't tried splashing the rubber tenga yet to fill it with other fluids, I just assumed it was only for cum...I just woke up and will try after I get my coffee and dung in a wad.) If I want other fluids I have to use the grinder or blender, which is always 20 units, but is also a good way to get stuff to flow steadily from a container, but also destroys the container you put there adding extra steps. That would be cool to have it pop the container off, making blenders and the grinder more functional. A button or click E to use would be nice on the blenders, but I feel like I've written too much and should maybe organize thoughts into other issues.

**Comment by GreenboxGames** [comment]
Actaully, I think I'm experiencing a bug that happens sometimes. I was playing the mod branch version from Unity and things were working fine. I opened the steam version and it's not like it was before. It's like sometimes, and I've had it happen enough to make me think it was on purpose. The trajectory even if I'm standing straight, double tapped R to make my hips straight, I toss and it goes straight up even though my cursor is forward... so it may be more of this glitch in the aim angle than the mechanics you are intending. I'll see if I can reproduce it in the Steam Build and then in Unity while testing...
